packages/core/r4b/data-types/identifier.ts

```diff
@@ -49,6 +50,25 @@ export type FhirIdentifierFormatOptions = {
   listFormatOptions?: Intl.ListFormatOptions | undefined;
 
   valueSetExpansions?: FhirCodeFormatOptions["valueSetExpansions"];
+
+  /**
+   * Optional pattern, or mapping of system and patterns. Formats the value according to the pattern when provided.
+   * When pattern = true, the value is formatted if the system is in the default pattern list (see {@link DEFAULT_SYSTEMS_PATTERNS}). Otherwise, the value is returned as is.
+   * The pattern consists of the following characters: '9' represents numbers, 'a' stands for letters, and '*' is used for both.
+   *
+   * @example
+   * // returns "123-456-7890"
+   * fhirIdentifierTypeAdapter.format(identifier, { pattern: "999-999-9999" })
+   * @example
+   * An us-mbi identifier will be formatted with the default system pattern
+   * // returns "1EG4-TE5-MK73"
+   * fhirIdentifierTypeAdapter.format(identifier, { pattern: true })
+   * @example
+   * Identifier with "https://fhir.nhs.uk/Id/nhs-number"  as system using the provided system:pattern mapping
+   * // returns "123 456 7890"
+   * fhirIdentifierTypeAdapter.format(identifier, { pattern: { "https://fhir.nhs.uk/Id/nhs-number": "999 999 9999" } })
+   */
+  pattern?: string | Record<string, string> | boolean | null | undefined;
```

Comment:
By default it should format using the default patterns.
I would suggest only accepting `false` as a value to disable the default patterns.


```suggestion
  pattern?: string | Record<string, string> | false | null | undefined;
```

Reply:
Default formatting using the default patterns would be a breaking change of the format function?

Reply:
Yes. In this case it's OK though, the impact is minimal and we are still in alpha :-)